Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif

```diff
@@ -0,0 +1,139 @@
+[MODEL: dataset=ibm_security_verify_raw]
+/*
+Mapping of 'management' event types.
+*/
+filter event_type = "management"
+| alter
+    src_user1 = arrayindex(regextract(data -> performedby_username, "(.+)\@") ,0), // The username can sometimes appear as a username and other times as an email.
+    src_user2 = arrayindex(regextract(data -> username, "(.+)\@") ,0),
+    mod = if(data -> modified = "[]", null, data -> modified)
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.target.application.name = data -> applicationname,
+    xdm.event.description = if(data -> cause != null AND data -> modified != null, concat(rtrim(data -> cause, "."), ". modified properties: ", mod), coalesce(data -> cause, data -> modified)), // Checks if 'cause' and 'modified' are both populated. If so, appends the cause with a list of modified properties. If not, returns whichever field is available.
+    xdm.event.original_event_type = data -> messageid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.source.user.identifier = if(data -> performedby IN("System", "shared_apikey"), null, data -> performedby),
+    xdm.source.application.name = coalesce(data -> performedby_clientname, if(data -> performedby IN("System", "shared_apikey"), data -> performedby)),
+    xdm.source.user.username = coalesce(src_user2, data -> username, src_user1, data -> performedby_username), //2 possible username fields, performedby_username and username.
+    xdm.source.user.upn = if(src_user2 != null, data -> username ,src_user1 != null, data -> performedby_username),
+    xdm.target.resource.id = data -> entitlementid,
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
```

Comment:
In 102 events the xdm.event.outcome should be only XDM_CONST.OUTCOME_SUCCESS or XDM_CONST.OUTCOME_FAILED